MoveKeys.actor.g.cpp.o -c /codebuild/output/src3439205401/src/github.com/apple/foundationdb/build_output/fdbserver/MoveKeys.actor.g.cpp
distcc[11255] ERROR: compile /codebuild/output/src3439205401/src/github.com/apple/foundationdb/build_output/fdbserver/MoveKeys.actor.g.cpp on distcc-c0afcb341c31fb8c.elb.us-west-2.amazonaws.com/96 failed
distcc[11255] (dcc_build_somewhere) Warning: remote compilation of '/codebuild/output/src3439205401/src/github.com/apple/foundationdb/build_output/fdbserver/MoveKeys.actor.g.cpp' failed, retrying locally
distcc[11255] Warning: failed to distribute /codebuild/output/src3439205401/src/github.com/apple/foundationdb/build_output/fdbserver/MoveKeys.actor.g.cpp to distcc-c0afcb341c31fb8c.elb.us-west-2.amazonaws.com/96, running locally instead
distcc[11255] ERROR: compile /codebuild/output/src3439205401/src/github.com/apple/foundationdb/build_output/fdbserver/MoveKeys.actor.g.cpp on localhost failed
/codebuild/output/src3439205401/src/github.com/apple/foundationdb/fdbserver/MoveKeys.actor.cpp:2967:117: error: use of undeclared identifier 'shardId'; did you mean 'srcId'?
                                emptyRangeActors.push_back(krmSetRangeCoalescing( &tr, serverKeysPrefixFor(id), range, allKeys, serverKeysValue(shardId)));
                                                                                                                                                ^~~~~~~
                                                                                                                                                srcId
/codebuild/output/src3439205401/src/github.com/apple/foundationdb/fdbserver/MoveKeys.actor.cpp:2851:6: note: 'srcId' declared here
        UID srcId;
            ^
/codebuild/output/src3439205401/src/github.com/apple/foundationdb/fdbserver/MoveKeys.actor.cpp:2967:117: error: use of undeclared identifier 'shardId'; did you mean 'srcId'?
                                emptyRangeActors.push_back(krmSetRangeCoalescing( &tr, serverKeysPrefixFor(id), range, allKeys, serverKeysValue(shardId)));
                                                                                                                                                ^~~~~~~
                                                                                                                                                srcId
/codebuild/output/src3439205401/src/github.com/apple/foundationdb/fdbserver/MoveKeys.actor.cpp:2851:6: note: 'srcId' declared here
        UID srcId;
            ^
2 errors generated.

--- a/fdbserver/MoveKeys.actor.cpp
+++ b/fdbserver/MoveKeys.actor.cpp
@@ -2940,7 +2940,7 @@ ACTOR Future<Void> removeKeysFromFailedServer(Database cx,
 							}
 						}
 
-						const UID shardId = newDataMoveId(
+						state UID shardId = newDataMoveId(
 						    deterministicRandom()->randomUInt64(), AssignEmptyRange::True, DataMoveType::LOGICAL);
 
 						// Assign the shard to teamForDroppedRange in keyServer space.
@@ -2957,20 +2957,22 @@ ACTOR Future<Void> removeKeysFromFailedServer(Database cx,
 							actors.push_back(
 							    krmSetRangeCoalescing(&tr, serverKeysPrefixFor(id), range, allKeys, serverKeysFalse));
 						}
+						wait(waitForAll(actors));
 
 						// Assign the shard to the new team as an empty range.
 						// Note, there could be data loss.
+						std::vector<Future<Void>> emptyRangeActors;
 						for (const UID& id : teamForDroppedRange) {
 							if (SERVER_KNOBS->SHARD_ENCODE_LOCATION_METADATA) {
-								actors.push_back(krmSetRangeCoalescing(
+								emptyRangeActors.push_back(krmSetRangeCoalescing(
 								    &tr, serverKeysPrefixFor(id), range, allKeys, serverKeysValue(shardId)));
 							} else {
-								actors.push_back(krmSetRangeCoalescing(
+								emptyRangeActors.push_back(krmSetRangeCoalescing(
 								    &tr, serverKeysPrefixFor(id), range, allKeys, serverKeysTrueEmptyRange));
 							}
 						}
 
-						wait(waitForAll(actors));
+						wait(waitForAll(emptyRangeActors));
 
 						TraceEvent trace(SevWarnAlways, "ShardLossAllReplicasDropShard", serverID);
 						trace.detail("Begin", it.key);

